Cython/Compiler/ExprNodes.py

```diff
@@ -1434,11 +1434,7 @@ def _analyse_name_as_type(name, pos, env):
         return type
 
     global_entry = env.global_scope().lookup(name)
-    if global_entry and global_entry.type and (
-            global_entry.type.is_extension_type
-            or global_entry.type.is_struct_or_union
-            or global_entry.type.is_builtin_type
-            or global_entry.type.is_cpp_class):
+    if global_entry and global_entry.is_type and global_entry.type:
```

Comment:
I can't say right now if there are any cases where we would like to disallow it, but I guess we can just wait until a user finds one and complains. :)